climakitae/derive_variables.py

```diff
@@ -19,24 +21,70 @@ def compute_hdd_cdd(t2, standard_temp=65):
         (hdd, cdd)
     """
 
-    # Subtract t2 from the standard reference temperature
-    deg_less_than_standard = standard_temp - t2
+    # Subtract t2 from the threshold inputs
+    hdd_deg_less_than_standard = hdd_threshold - t2
+    cdd_deg_less_than_standard = cdd_threshold - t2
 
     # Compute HDD: Find positive difference (i.e. days < 65 degF)
-    hdd = deg_less_than_standard.where(
-        deg_less_than_standard > 0, 0
+    hdd = hdd_deg_less_than_standard.where(
+        hdd_deg_less_than_standard > 0, 0
     )  # Replace negative values with 0
     hdd.name = "Heating Degree Days"
+    hdd.attrs["hdd_threshold"] = (
+        str(hdd_threshold) + "degF"
+    )  # add attribute of threshold value
 
     # Compute CDD: Find negative difference (i.e. days > 65 degF)
-    cdd = (-1) * deg_less_than_standard.where(
-        deg_less_than_standard < 0, 0
+    cdd = (-1) * cdd_deg_less_than_standard.where(
+        cdd_deg_less_than_standard < 0, 0
     )  # Replace positive values with 0
     cdd.name = "Cooling Degree Days"
+    cdd.attrs["cdd_threshold"] = (
+        str(cdd_threshold) + "degF"
+    )  # add attribute of threshold value
 
     return (hdd, cdd)
 
 
+def compute_hdh_cdh(t2, hdh_threshold=65, cdh_threshold=65):
+    """Compute heating degree hours (HDH) and cooling degree hours (CDH)
+
+    Parameters
+    -----------
+    t2: xr.DataArray
+        Air temperature at 2m gridded data
+    hdh_threshold: int, optional
+        Standard temperature in Fahrenheit. Default to 65 degF
+    cdh_threshold: int, optional
+        Standard temperature in Fahrenheit. Default to 65 degF
+
+    Returns
+    -------
+    tuple of xr.DataArray
+        (hdh, cdh)
+    """
+
+    # Calculate heating and cooling hours
+    cooling_hours = t2.where(
+        t2 > hdh_threshold
+    )  # temperatures above threshold, require cooling
+    heating_hours = (-1) * t2.where(
+        t2 < cdh_threshold
+    )  # temperatures below threshold, require heating
+
+    # Compute CDH: count number of hours and resample to daily (max 24 value)
+    cdh = cooling_hours.resample(time="1D").count(dim="time").squeeze()
+    cdh.name = "Cooling Degree Hours"
+    cdh.attrs["cdh_threshold"] = str(cdh_threshold) + "degF"
```

Comment:
Would recommend adding a space here (and to the hdh, hdd, and cdd attributes of the same nature) so that it results in "60 degF" vs "60degF" (or whatever the threshold temp is)